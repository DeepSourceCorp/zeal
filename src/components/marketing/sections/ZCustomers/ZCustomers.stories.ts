import ZCustomers from './ZCustomers.vue'

export default {
  title: 'Zeal Marketing/Customers section',
  component: ZCustomers,
  excludeStories: /.*Data$/
}

export const Basic = () => ({
  components: { ZCustomers },
  data() {
    return {
      customers: [
        {
          title: 'Slack1',
          slug: 'Slack1',
          image: 'https://i.imgur.com/mu7Vh6o.png',
          url: ''
        },
        {
          title: 'Slack2',
          slug: 'Slack2',
          image: 'https://i.imgur.com/BxxXp7v.png',
          url: ''
        },
        {
          title: 'Nasa',
          slug: 'Nasa',
          image: 'https://i.imgur.com/46V02HC.png',
          url: ''
        },
        {
          title: 'Sas',
          slug: 'Sas',
          image: 'https://i.imgur.com/v7zouKN.png',
          url: ''
        },
        {
          title: 'Dgraph',
          slug: 'Dgraph',
          image: 'https://i.imgur.com/ZJBUi4D.png',
          url: ''
        },
        {
          title: 'Qubole',
          slug: 'Qubole',
          image: 'https://i.imgur.com/O9bfIqK.png',
          url: ''
        },
        {
          title: 'Intel',
          slug: 'Intel',
          image: 'https://i.imgur.com/4uKLbka.png',
          url: ''
        },
        {
          title: 'Okteto',
          slug: 'Okteto',
          image: 'https://i.imgur.com/9gqeCeh.png',
          url: ''
        }
      ]
    }
  },
  template: `<div class='padded-container'>
      <z-customers :customers="customers">
        <h2 slot="heading">Used by developers from</h2>
      </z-customers>
    </div>`
})
