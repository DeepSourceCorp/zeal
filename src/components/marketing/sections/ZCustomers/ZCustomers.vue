<template>
  <section class="w-full-screen">
    <div class="text-center text-vanilla-100 text-sm uppercase font-bold tracking-widest mb-8">
      <slot name="heading"></slot>
    </div>
    <div class="grid grid-cols-2 md:grid-cols-4 grid-row-4 md:grid-rows-2 border-t border-ink-200">
      <span
        v-for="(customer, index) in customers"
        :key="customer.slug"
        class="flex items-center border-b border-ink-200 h-20 md:h-40"
        :class="[`${!isFourthElement(index) ? 'border-r even:border-r-0 md:even:border-r' : ''}`]"
      >
        <img
          :src="customer.image"
          :alt="customer.title"
          :title="customer.title"
          class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"
        />
      </span>
    </div>
  </section>
</template>

<script lang="ts">
import Vue from 'vue'

const MAX_CUSTOMERS = 8

export default Vue.extend({
  name: 'ZCustomers',
  props: {
    customers: {
      type: Array,
      required: true,
      validator: arr => arr.length <= MAX_CUSTOMERS
    }
  },
  data() {
    return {}
  },
  methods: {
    isFourthElement(index: number) {
      return (index + 1) % 4 === 0
    }
  }
})
</script>
