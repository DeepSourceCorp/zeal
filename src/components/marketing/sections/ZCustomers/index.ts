import ZCustomers from './ZCustomers.vue'

export default ZCustomers
