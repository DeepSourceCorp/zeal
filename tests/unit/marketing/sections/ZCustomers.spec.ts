import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZCustomers from '../../../../src/components/marketing/sections/ZCustomers'

const customers = [
  {
    title: 'Slack1',
    slug: 'Slack1',
    image: 'https://i.imgur.com/mu7Vh6o.png',
    url: ''
  },
  {
    title: 'Slack2',
    slug: 'Slack2',
    image: 'https://i.imgur.com/BxxXp7v.png',
    url: ''
  },
  {
    title: 'Nasa',
    slug: 'Nasa',
    image: 'https://i.imgur.com/46V02HC.png',
    url: ''
  },
  {
    title: 'Sas',
    slug: 'Sas',
    image: 'https://i.imgur.com/v7zouKN.png',
    url: ''
  },
  {
    title: 'Dgraph',
    slug: 'Dgraph',
    image: 'https://i.imgur.com/ZJBUi4D.png',
    url: ''
  },
  {
    title: 'Qubole',
    slug: 'Qubole',
    image: 'https://i.imgur.com/O9bfIqK.png',
    url: ''
  },
  {
    title: 'Intel',
    slug: 'Intel',
    image: 'https://i.imgur.com/4uKLbka.png',
    url: ''
  },
  {
    title: 'Okteto',
    slug: 'Okteto',
    image: 'https://i.imgur.com/9gqeCeh.png',
    url: ''
  }
]

describe('Customers section', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZCustomers, {
        localVue,
        ...options
      })
    }
  })
  it('renders customers section when the component is mounted', () => {
    let mountedInstance = mountFunction({
      propsData: {
        customers
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
  })

  it('renders the heading correctly', () => {
    let mountedInstance = mountFunction({
      propsData: {
        customers
      },
      slots: {
        heading: 'This is heading'
      }
    })
    expect(mountedInstance.text()).toContain('This is heading')
  })

  it('renders eight logos', () => {
    let mountedInstance = mountFunction({
      propsData: {
        customers
      }
    })
    expect(mountedInstance.findAll('span.border-b').length).toEqual(8)
  })

  it('renders every logo, except the fourth logo, with right border', () => {
    let mountedInstance = mountFunction({
      propsData: {
        customers
      }
    })
    expect(mountedInstance.findAll('span.border-b').at(0).classes('border-r')).toBe(true)
    expect(mountedInstance.findAll('span.border-b').at(3).classes('border-r')).toBe(false)
  })
})
