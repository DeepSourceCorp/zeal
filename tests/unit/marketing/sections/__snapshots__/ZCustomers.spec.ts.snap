// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customers section renders customers section when the component is mounted 1`] = `
<section class="w-full-screen">
  <div class="text-center text-vanilla-100 text-sm uppercase font-bold tracking-widest mb-8"></div>
  <div class="grid grid-cols-2 md:grid-cols-4 grid-row-4 md:grid-rows-2 border-t border-ink-200"><span class="flex items-center border-b border-ink-200 h-20 md:h-40 border-r even:border-r-0 md:even:border-r"><img src="https://i.imgur.com/mu7Vh6o.png" alt="Slack1" title="Slack1" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span><span class="flex items-center border-b border-ink-200 h-20 md:h-40 border-r even:border-r-0 md:even:border-r"><img src="https://i.imgur.com/BxxXp7v.png" alt="Slack2" title="Slack2" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span><span class="flex items-center border-b border-ink-200 h-20 md:h-40 border-r even:border-r-0 md:even:border-r"><img src="https://i.imgur.com/46V02HC.png" alt="Nasa" title="Nasa" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span><span class="flex items-center border-b border-ink-200 h-20 md:h-40"><img src="https://i.imgur.com/v7zouKN.png" alt="Sas" title="Sas" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span><span class="flex items-center border-b border-ink-200 h-20 md:h-40 border-r even:border-r-0 md:even:border-r"><img src="https://i.imgur.com/ZJBUi4D.png" alt="Dgraph" title="Dgraph" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span><span class="flex items-center border-b border-ink-200 h-20 md:h-40 border-r even:border-r-0 md:even:border-r"><img src="https://i.imgur.com/O9bfIqK.png" alt="Qubole" title="Qubole" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span><span class="flex items-center border-b border-ink-200 h-20 md:h-40 border-r even:border-r-0 md:even:border-r"><img src="https://i.imgur.com/4uKLbka.png" alt="Intel" title="Intel" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span><span class="flex items-center border-b border-ink-200 h-20 md:h-40"><img src="https://i.imgur.com/9gqeCeh.png" alt="Okteto" title="Okteto" class="mx-auto object-contain h-10 w-24 md:h-auto md:w-auto"></span></div>
</section>
`;
